Dockerfile

```diff
@@ -1,79 +1,91 @@
+# syntax = docker/dockerfile-upstream:master-labs
+
 FROM alpine:3.15
 
-RUN apk --no-cache add \
+ARG PHP_VERSION=php8
+
+RUN <<EOF
+    set -x
+    apk --no-cache add \
     samba-common-tools samba-client samba-server \
     bash ncurses curl python3 gcc libc-dev perl make rpcgen file ssmtp supervisor gnutls-dev zlib-dev rsyslog \
-    php8-cli php8-pdo_mysql php8-intl php8-mbstring php8-intl php8-mysqlnd php8-json php8-pcntl rsync lsof sysstat findutils gzip patch \
-    && ln -s /usr/bin/php8 /usr/bin/php
-
-# SSMTP (to be able to send emails)
-COPY ssmtp.conf /etc/ssmtp/ssmtp.conf
-RUN echo "hostname=$(hostname).home.danslereseau.com" >> /etc/ssmtp/ssmtp.conf
+    $PHP_VERSION-cli $PHP_VERSION-pdo_mysql $PHP_VERSION-intl $PHP_VERSION-mbstring $PHP_VERSION-intl $PHP_VERSION-mysqlnd $PHP_VERSION-json $PHP_VERSION-pcntl rsync lsof sysstat findutils gzip patch
+    ln -s /usr/bin/$PHP_VERSION /usr/bin/php
+EOF
 
 # Setup Greyhole for Samba
-ARG GREYHOLE_VERSION=master
```

Comment:
I was using the `GREYHOLE_VERSION` build argument to build an image for a specific version.
See https://github.com/gboudreau/Greyhole/blob/master/build/build_greyhole.sh#L230
This is used when I package a new version of Greyhole; I build Docker images too, and push those to Dockerhub.

Not sure it works, but can we do this?

    ARG GREYHOLE_VERSION=master
    ADD --link --keep-git-dir=false https://github.com/gboudreau/Greyhole.git#$GREYHOLE_VERSION /Greyhole-master

Reply:
Heya, 

So master would be a branch, but that could change at any time you make a commit, whilst with the version, it's static, and the dockerfile will always pull a specific, static version from GitHub, effectively working on the ref vs the branch. 

What we can do, is change is so it's:

`https://github.com/gboudreau/Greyhole.git$GREYHOLE_VERSION`

With GREYHOLE_VERSION either set to a particular version (commit), or empy, in which case, it'd grab the "master" branch. 

https://docs.docker.com/engine/reference/builder/#adding-a-git-repository-add-git-ref-dir

Reply:
`ARG GREYHOLE_VERSION=master` just defines the default value for the build argument, so when not specifying a `--build-arg "GREYHOLE_VERSION=$VERSION"`, it uses master, but otherwise, it uses the specified version (tag).
Also, I think you are missing a `#` character in there, between the git repo URL, and the tag to pull.

Reply:
Aye, unfortunately I got distracted whilst editing my comment :P. 

My thinking was: `https://github.com/gboudreau/Greyhole.git$GREYHOLE_VERSION` with a buildarg set as a variable to empty, and you can add a commit version number if you so wish. 

That way, if you only do `https://github.com/gboudreau/Greyhole.git` it grabs "master" which would be the default behaviour unless you so wish to change with the build arg.

Reply:
Nope. I was wrong. 

Will update the Dockerfile now to reflect what you require!

Reply:
Just pushed https://github.com/gboudreau/samba-greyhole-docker/pull/1/commits/f4e50dd138c792f0e8c5926eaaa4ba51f4f979e8 - does this do what you're after? 

I've also checked out Alpine 3.16 and this does not build, seems there's a potential issue w/ php8.